### PR TITLE
CSP-764 Implements GET CalendarEvent details endpoint

### PR DESCRIPTION
**Pull request by SawNawDfE**
- Add entities, repository, models, data context and other common classes related to getting calendar events 
- Update Member entity and MemberConfiguration

### CODE REVIEW

File: src/SFA.DAS.AANHub.Data/Configuration/MemberConfiguration.cs
Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Data/Configuration/MemberConfiguration.cs:34]: @DevenKShah Can we delete these now?

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs
Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:16]: Since only live attendees should be included, this should probably be:
` .Include(x => x.Attendees.Where(a => a.IsActive)`

Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:16]: Correct

Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs]: Done

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs
Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:40]: Here I'm logging both the error and the user ID of the request, since that would make it easier to track errors. The calendar event ID is already being logged.

File: src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs
Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs:12]: Since GUIDs cannot be null, I've removed this

Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs:12]: Since GUIDs cannot be null, I've removed this

File: src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs
Comment (SawNawDfE) [inline src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs:13]: If the X-RequestedByUser header is missing, _both_ the "empty" error message and the "member ID not found (in the database)" message are displayed. We can use `When` and `Otherwise` to make it display only one.

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEvent/GetCalendarEventByIdQueryResult.cs
Comment (MarkFCain) [inline src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEvent/GetCalendarEventByIdQueryResult.cs:29]: All these fields look exactly the same as CalendarEvent? Can this inherit from CalendarEvent instead? (or maybe both based on common parent)? Have a look, I do know that sometimes this causes implicit operators and other mappers to fall apart though, so use your judgement to decide whether to address or not.

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs
Comment (MarkFCain) [inline src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs:84]: don't bother with space lines between the result checkers.

File: src/SFA.DAS.AANHub.Application/Models/EventGuestModel.cs
Comment (MarkFCain) [inline src/SFA.DAS.AANHub.Application/Models/EventGuestModel.cs:7]: Can we have some coverage on this implicit operator? Thanks

File: src/SFA.DAS.AANHub.Application/Models/AttendeeModel.cs
Comment (MarkFCain) [inline src/SFA.DAS.AANHub.Application/Models/AttendeeModel.cs:7]: Can we have some coverate on this implicit operator? thanks

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:40]: I thought those details would be in the header anyway? This seems like a lot of logging that clutters this controller.  We don't usually do this.   I'll add a note here so @DevenKShah can comment, and will defer to his decision

File: src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs
Comment (MarkFCain) [reply src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs:13]: Not sure this matters much, but yes it would be more sensible that way

File: src/SFA.DAS.AANHub.Data/Configuration/CalendarEventConfiguration.cs
Comment (MarkFCain) [inline src/SFA.DAS.AANHub.Data/Configuration/CalendarEventConfiguration.cs:7]: You can add a '[ExcludeFromCodeCoverage]' to this

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs
Comment (MarkFCain) [inline src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:6]: you can add a '[ExcludeFromCodeCoverage]' to this

**Review by MarkFCain — COMMENTED** [review]
pick up the coverage, and fix code smells, and I will approve

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:40]: I agree with Mark, there is no need to log any of these validation failures. Also these are not errors and should not be logged as errors. Please refer [WoW ](https://skillsfundingagency.atlassian.net/wiki/spaces/NDL/pages/3929210881/AAN+Dev+Team+Ways+of+working#Error) for logging tips.

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:12]: Can you capture the query that is being generated here please, I want to check if this is going to be inefficient to run when there is lot more data.

File: src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs:13]: @SawNawDfE you can use Stop Cascade mode to stop validating after first failure... there are many examples of this

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:6]: Attribute added

File: src/SFA.DAS.AANHub.Data/Configuration/CalendarEventConfiguration.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Data/Configuration/CalendarEventConfiguration.cs:7]: Attribute added

File: src/SFA.DAS.AANHub.Application/Models/AttendeeModel.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Application/Models/AttendeeModel.cs:7]: Test coverage added

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs:84]: Space removed

File: src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Application/Common/Validators/RequestedByMemberId/RequestedByMemberIdValidator.cs:13]: Stop cascade used

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:40]: Invalid response logging removed

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:30]: Can the header be a constant here as we will be using this in many places.

File: src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEvent/GetCalendarEventByIdQueryResult.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEvent/GetCalendarEventByIdQueryResult.cs]: The result is the response body  of the controller and result is using Entity types which can cause cyclical serialisation issue. Entities needs to be converted in to models in the handler or in the controller.

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Queries/GetCalendarEventByIdQueryHandlerTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Queries/GetCalendarEventByIdQueryHandlerTests.cs:32]: Naming convention is <Action>_<Condition>_<Result> as explained in the Wow document. Suggest rename to `Handle_CalendarEventFound_ReturnsCalendarEvent`

File: src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Queries/GetCalendarEventByIdQueryValidatorTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/CalendarEvents/Queries/GetCalendarEventByIdQueryValidatorTests.cs:16]: ValidateCalendarId_Invalid_FailsValidation

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberQueryValidatorTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberQueryValidatorTests.cs]: Use `WithMessage` instead of `WithErrorCode` we are not using error codes anywhere.

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:15]: Nice, not efficient query TBH.

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberQueryValidatorTests.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberQueryValidatorTests.cs]: Replaced

File: src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs:18]: I found an existing constant [here](https://github.com/SkillsFundingAgency/das-aan-hub-api/blob/a4d4e4f5b02f6c986d95a447bb221b9cbef66880/src/SFA.DAS.AANHub.Api/Common/Constants.cs#L7)  Just change the value to match yours and also change the class name from `PostRequestHeaders` to `RequestHeaders`

Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Api/Controllers/CalendarEventsController.cs]: Done, please could you re-review?

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs:28]: can add requestedMemberId check here as well.

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs:15]: @DevenKShah We could consider using split queries to reduce the joins: 
https://learn.microsoft.com/en-us/ef/core/querying/single-split-queries

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs
Comment (SawNawDfE) [reply src/SFA.DAS.AANHub.Api.UnitTests/Controllers/CalendarEventsControllerTests/CalendarEventsControllerGetTests.cs]: Done, and I've also tried to do some housekeeping by shortening the test names and updating NUnit/AutoFixture packages - please re-review and approve if happy

**Review by MarkFCain — DISMISSED** [review]

**Comment by SawNawDfE** [comment]
Closing because the work has been re-done on the branch CSP-764-Reset

**Pull request closed without merging**
